packages/hooks/tests/memo.rs

```diff
@@ -68,3 +68,111 @@ async fn memo_updates() {
         println!("Signal: {signal:?}");
     }
 }
+
+#[tokio::test]
+async fn use_memo_with_use_effect() {
+    use dioxus::prelude::*;
+    use futures_util::future::select;
+    use std::{cell::RefCell, collections::VecDeque, pin::pin, rc::Rc};
+    use tokio::sync::mpsc;
+
+    let (action_sender, mut action_receiver) = mpsc::unbounded_channel::<Action>();
+
+    #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+    enum Action {
+        Write(Option<usize>, Option<usize>),
+        Read(usize, usize),
+    }
+
+    #[derive(Clone)]
+    struct ActionSender(Rc<RefCell<mpsc::UnboundedSender<Action>>>);
+
+    #[component]
+    fn App() -> Element {
+        let action_sender = use_context::<ActionSender>();
+        let mut a = use_signal(|| 0);
+        let mut b = use_signal(|| 0);
+        let mut counter = use_signal(|| 0);
+
+        use_effect({
+            let action_sender = action_sender.clone();
+            move || {
+                let count = counter();
+                let action = match count {
+                    0 => Action::Write(Some(0), None),
+                    1 => Action::Write(None, Some(0)),
+                    2 => Action::Write(Some(0), Some(0)),
+                    3 => Action::Write(Some(1), None),
+                    4 => Action::Write(None, Some(1)),
+                    5 => Action::Write(Some(1), Some(1)),
+                    6 => Action::Write(Some(2), Some(2)),
+                    7 => Action::Write(Some(0), Some(0)),
+                    _ => return,
+                };
+                action_sender.0.borrow_mut().send(action).unwrap();
+                if let Action::Write(Some(n), _) = action {
+                    *a.write() = n;
+                }
+                if let Action::Write(_, Some(n)) = action {
+                    *b.write() = n;
+                }
+                *counter.write() = count + 1;
+            }
+        });
+
+        let tuple = use_memo(move || (a(), b()));
+
+        use_effect(move || {
+            let (a, b) = tuple();
+            action_sender
+                .0
+                .borrow_mut()
+                .send(Action::Read(a, b))
+                .unwrap();
+        });
+
+        None
```

Comment:
Ok, interesting, if you change this line to:

```suggestion
        rsx! {"{tuple:?}"}
```

The test passes.